sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -103,6 +106,13 @@ object DataType {
 
   def fromJson(json: String): DataType = parseDataType(parse(json))
 
+  // Until Spark-2.1, we use json strings for defining schemas in user-facing APIs.
+  // Since we add an user-friendly API in the DDL parser, we employ DDL formats for the case.
+  // To keep back-compatibility, we use `fromJson` first, and then try the new API.
+  def fromString(text: String): DataType = {
```

Comment:
Thanks for the valuable comments!
Yea, I basically agree with @HyukjinKwon idea. Also, `fromDdl` is better than `fromString` to me as @viirya suggested. I'll update soon.